docs/python.rst

```diff
@@ -50,35 +50,50 @@ For higher dimensions, we would specify point locations in more than one dimensi
+    # the 2D transform outputs f array of shape (N1, N2)
+    f = finufftpy.nufft2d1(x, y, c, (N1, N2))
 
-See ``python/examples/simple2d1.py`` for the demo code that includes a basic math test (useful to check both the math and the indexing). We can also go the other way, from uniform to non-uniform points, using a type 2 transform:
+See ``python/examples/simple2d1.py`` for the demo code that includes a basic math test (useful to check both the math and the indexing).
+
+We can also go the other way, from uniform to non-uniform points, using a type 2 transform:
 
 .. code-block:: python
 
     # coefficients
-    f = (np.random.standard_normal(size=(N1,N2))
-         + 1J * np.random.standard_normal(size=(N1,N2)))
+    f = (np.random.standard_normal(size=(N1, N2))
+         + 1J * np.random.standard_normal(size=(N1, N2)))
 
-    # calcualate the 2D type 2 transform
+    # calculate the 2D type 2 transform
     c = finufftpy.nufft2d2(x, y, f)
 
-Now the output is a complex vector of length ``M`` approximating (2) in :ref:`math`, that is the adjoint (but not inverse) of (1). (Note that the default in the python interface is that sign in the exponential is negative for type 2.)
+Now the output is a complex vector of length ``M`` approximating (2) in :ref:`math`, that is the adjoint (but not inverse) of (1). (Note that the default sign in the exponential is negative for type 2 in the Python interface.)
 
 In addition to tolerance ``eps``, we can adjust other options for the transform.
 These are listed in :ref:`opts` and are specified as keyword arguments in the Python interface.
-For example, to change the mode ordering to FFT style (that is, in each dimension ``Ni=N1`` or ``N2``, the indices go from ``0`` to ``Ni/2-1``, then from ``-Ni/2`` to ``-1``, since each ``Ni`` is even), we call
+For example, to change the mode ordering to FFT style (that is, in each dimension ``Ni = N1`` or ``N2``, the indices go from ``0`` to ``Ni // 2 - 1``, then from ``-Ni // 2`` to ``-1``, since each ``Ni`` is even), we call
+
+.. code-block:: python
+
+    f = finufftpy.nufft2d1(x, y, c, (N1, N2), modeord=1)
+
+We can also specify a preallocated output array using the ``out`` keyword argument.
+This would be done by
 
 .. code-block:: python
 
-    f = finufftpy.nufft2d1(x, y, c, (N1,N2), modeord=1)
+    # allocate the output array
+    f = np.empty((N1, N2), dtype='complex128')
+
+    # calculate the transform
+    finufftpy.nufft2d1(x, y, c, out=f)
+
+In this case, we do not need to specify the output shape since it can be inferred from ``f``.
 
-Note that the above functions are all vectorized, which means that they can take multiple inputs stacked along the last dimension (that is, in column-major order) and process them simultaneously.
+Note that the above functions are all vectorized, which means that they can take multiple inputs stacked along the first dimension (that is, in row-major order) and process them simultaneously.
```

Comment:
you're right -thx.

Reply:
actually I'd fixed this too but forgot to push it.

Reply:
No problem. Was my mistake to begin with…